src/Temporalio/Bridge/NativeInvokeableClass.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Temporalio.Bridge
+{
+    /// <summary>
+    /// Extend this class to help with making a class that has callbacks which are invoked by Rust.
+    /// </summary>
+    /// <typeparam name="T">The native type that holds the function ptrs for callbacks to C#.</typeparam>
+    internal class NativeInvokeableClass<T>
+    where T : unmanaged
+    {
+        private readonly List<GCHandle> handles = new();
+
+        /// <summary>
+        /// Gets the pointer to the native callback holder.
+        /// </summary>
+        internal unsafe T* Ptr { get; private set; }
+
+        /// <summary>
+        /// Pin the native type in memory and add it to the handle list. Call this after adding
+        /// the callbacks via <see cref="FunctionPointer"/>. Also adds `this` to the handle list.
+        /// </summary>
+        /// <param name="value">The native type to pin.</param>
+        internal void PinCallbackHolder(T value)
```

Comment:
Technically these calls only invoked by subclasses should be `private protected` but not a big deal